lib/taskana-core-test/src/test/java/acceptance/task/create/CreateTaskWithAttachmentAccTest.java

```diff
@@ -0,0 +1,201 @@
+import pro.taskana.testapi.builder.TaskAttachmentBuilder;
+import pro.taskana.testapi.builder.TaskBuilder;
+import pro.taskana.testapi.builder.WorkbasketAccessItemBuilder;
+import pro.taskana.testapi.security.WithAccessId;
+import pro.taskana.workbasket.api.WorkbasketPermission;
+import pro.taskana.workbasket.api.WorkbasketService;
+import pro.taskana.workbasket.api.models.WorkbasketSummary;
+
+/**
+ * Acceptance test for all "create task" scenarios that involve {@linkplain
+ * pro.taskana.task.api.models.Attachment Attachments}.
+ */
+@TaskanaIntegrationTest
+class CreateTaskWithAttachmentAccTest {
+  @TaskanaInject TaskService taskService;
+  @TaskanaInject WorkbasketService workbasketService;
+  @TaskanaInject ClassificationService classificationService;
+
+  ClassificationSummary defaultClassificationSummary;
+  WorkbasketSummary defaultWorkbasketSummary;
+  ObjectReference defaultObjectReference;
+  Attachment defaultAttachment;
+  Task defaultTaskWithAttachment;
+
+  @WithAccessId(user = "businessadmin")
+  @BeforeAll
+  void setup() throws Exception {
+    defaultClassificationSummary =
+        defaultTestClassification()
+            .serviceLevel("P2D")
+            .buildAndStoreAsSummary(classificationService);
+    defaultWorkbasketSummary = defaultTestWorkbasket().buildAndStoreAsSummary(workbasketService);
+
+    WorkbasketAccessItemBuilder.newWorkbasketAccessItem()
+        .workbasketId(defaultWorkbasketSummary.getId())
+        .accessId("user-1-1")
+        .permission(WorkbasketPermission.OPEN)
+        .permission(WorkbasketPermission.READ)
+        .permission(WorkbasketPermission.APPEND)
+        .buildAndStore(workbasketService);
+    defaultObjectReference = defaultTestObjectReference().build();
+    defaultAttachment =
+        TaskAttachmentBuilder.newAttachment()
+            .classificationSummary(defaultClassificationSummary)
+            .objectReference(defaultObjectReference)
+            .build();
+    defaultTaskWithAttachment =
+        TaskBuilder.newTask()
+            .primaryObjRef(defaultObjectReference)
+            .workbasketSummary(defaultWorkbasketSummary)
+            .classificationSummary(defaultClassificationSummary)
+            .attachments(defaultAttachment)
+            .buildAndStore(taskService, "user-1-1");
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_SetTaskIdOfAttachmentCorrectly_WhenCopyingAttachment() throws Exception {
+    Attachment copiedAttachment =
+        taskService.getTask(defaultTaskWithAttachment.getId()).getAttachments().get(0).copy();
+    Task taskToCreate = taskService.newTask(defaultWorkbasketSummary.getId());
+    taskToCreate.setClassificationKey(defaultClassificationSummary.getKey());
+    taskToCreate.setPrimaryObjRef(defaultObjectReference);
+    taskToCreate.addAttachment(copiedAttachment);
+    Task result = taskService.createTask(taskToCreate);
+
+    assertThat(result.getAttachments().get(0).getTaskId()).isEqualTo(result.getId());
+  }
```

Comment:
This does not test TSK-1824 since the `copiedAttachment` does not contain a different taskId.
Please don't copy the Attachment. Just reuse the Attachment from the `defaultTaskWithAttachment`

Reply:
Ok :)